tests/union_underlying_type.fbs

```diff
@@ -0,0 +1,15 @@
+namespace union_underlying_type;
+
+table A {}
+table B {}
+
+union AB : int {
+  A,
```

Comment:
to generate less "noise", both of these fields could strings, meaning you wouldn't need the two table decls, which would shrink the amount of generated code a fair bit. But I guess that is not supported by all languages yet?

Reply:
Alternatively, we could stick this union in an existing schema (even the main one), such that it can make use of existing tables, and sit in an existing root table, such that there is less new stuff to generate.

Reply:
I think we should keep it separate from other tests - we have a simple implementation, built for every language, which is explicitly tested by `test.cpp`.

Reply:
@aardappel what do you think about keeping it separated?

Reply:
I don't see why it should be separate if we're implementing it for every language. But maybe @vglavnyy @dbaileychess @CasperN have an opinion.

See also my comments above about missing tests for other languages, docs, json parsing etc.